packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -987,139 +1090,65 @@ class GoTrueClient {
     return OAuthResponse(provider: provider, url: oauthUrl);
   }
 
-  void _saveSession(Session session) async {
+  /// set currentSession and currentUser
+  void _saveSession(Session session) {
     _currentSession = session;
     _currentUser = session.user;
-    final expiresAt = session.expiresAt;
-
-    if (_autoRefreshToken && expiresAt != null) {
-      _refreshTokenTimer?.cancel();
-
-      final timeNow = (DateTime.now().millisecondsSinceEpoch / 1000).round();
-      final expiresIn = expiresAt - timeNow;
-      final refreshDurationBeforeExpires = expiresIn > 60 ? 60 : 1;
-      final nextDuration = expiresIn - refreshDurationBeforeExpires;
-      try {
-        if (nextDuration > 0) {
-          _refreshTokenRetryCount = 0;
-          final timerDuration = Duration(seconds: nextDuration);
-          _setTokenRefreshTimer(timerDuration);
-        } else {
-          await _callRefreshToken();
-        }
-      } catch (e) {
-        // Catch any error, because in this case they should be handled by listening to [onAuthStateChange]
-      }
-    }
-  }
-
-  void _setTokenRefreshTimer(
-    Duration timerDuration, {
-    String? refreshToken,
-    String? accessToken,
-  }) {
-    _refreshTokenTimer?.cancel();
-    _refreshTokenRetryCount++;
-    if (_refreshTokenRetryCount < Constants.maxRetryCount) {
-      _refreshTokenTimer = Timer(timerDuration, () async {
-        try {
-          await _callRefreshToken(
-            refreshToken: refreshToken,
-            accessToken: accessToken,
-            ignorePendingRequest: true,
-          );
-        } catch (_) {
-          // Catch any error, because in this case they should be handled by listening to [onAuthStateChange]
-        }
-      });
-    } else {
-      throw AuthException('Access token refresh retry limit exceeded.');
-    }
   }
 
   void _removeSession() {
     _currentSession = null;
     _currentUser = null;
-    _refreshTokenRetryCount = 0;
-
-    _refreshTokenTimer?.cancel();
   }
 
   /// Generates a new JWT.
   ///
   /// To prevent multiple simultaneous requests it catches an already ongoing request by using the global [_refreshTokenCompleter].
   /// If that's not null and not completed it returns the future of the ongoing request.
   ///
-  /// To call [_callRefreshToken] during a running request [ignorePendingRequest] is used to bypass that check.
-  ///
   /// When a [ClientException] occurs [_setTokenRefreshTimer] is used to schedule a retry in the background, which emits the result via [onAuthStateChange].
```

Comment:
This comment is outdated now. This method doesn't retry anything now and just reports any error and may sign out.